### PR TITLE
Attempt to fix the stake modifier mismatch.

### PR DESCRIPTION
**Pull request by tomasbrod**
Note: ~this patch was NOT tested.~ Syncs OK, but can't reproduce the original issue.
Before merging, try if it is possible to sync from zero using bootstrap (not snapshot!).
#572

### CODE REVIEW

**Comment by denravonska** [comment]
Do we have to test in prod or is it enough to test on testnet?

**Comment by tomasbrod** [comment]
The blockHash there is for prod block 1010000, it must be done in prod.
While we are at it, even if this has no effect in testnet, it should be simultaneously tried in testnet. Just to be sure.

**Comment by tomasbrod** [comment]
barton26: <rjak> brod: Your PR (581) for 572 is very unlikely to work because ComputeNextStakeModifier calls SelectBlockFromCandidates.

**Comment by denravonska** [comment]
Tested on both prod and testnet using bootstrap. None of them got stuck. On the other hand, the warning doesn't show up in the log.

**Comment by tomasbrod** [comment]
Somewhat I can't trigger this error anymore. I guess there is some high time dependency. Anyway,
this PR just ensures that the modifier is correct (constant) on that problematic block.
TLDR: this version can sync from bootstrap.
